common/src/test/java/io/netty/util/concurrent/DefaultThreadFactoryTest.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2013 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.util.concurrent;
+
+import org.junit.Test;
+
+import java.security.Permission;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+public class DefaultThreadFactoryTest {
+
+    // test that ensures DefaultThreadGroup is not sticky about thread groups by default
+    @Test
+    public void testStickyThreadGroups() throws InterruptedException {
+
+        final AtomicReference<DefaultThreadFactory> factory = new AtomicReference<DefaultThreadFactory>();
+        final AtomicReference<ThreadGroup> capturedGroup = new AtomicReference<ThreadGroup>();
+        final AtomicReference<Thread.UncaughtExceptionHandler> capturedUh =
+                new AtomicReference<Thread.UncaughtExceptionHandler>();
+
+        final ThreadGroup brother = new ThreadGroup("brother");
+
+        final Thread first = new Thread(brother, new Runnable() {
+            @Override
+            public void run() {
+                factory.set(new DefaultThreadFactory(this.getClass()));
+            }
+        });
+        first.start();
+        first.join();
+
+        final ThreadGroup sister = new ThreadGroup("sister");
+
+        final Thread second = new Thread(sister, new Runnable() {
+            @Override
+            public void run() {
+                final Thread t = factory.get().newThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        capturedGroup.set(Thread.currentThread().getThreadGroup());
+                        capturedUh.set(Thread.currentThread().getUncaughtExceptionHandler());
+                    }
+                });
+                t.start();
+                try {
+                    t.join();
+                } catch (InterruptedException e) {
+                    fail(e.toString());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2186)